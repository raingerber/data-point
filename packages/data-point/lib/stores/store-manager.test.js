/* eslint-env jest */

const _ = require('lodash')
const ObjectStoreManager = require('./store-manager')

function createManager () {
  return {
    store: new Map()
  }
}

describe('add', () => {
  test('It should throw error with information if item already exist', () => {
    const mgr = createManager()
    const errorInfoCb = () => ({
      name: 'Foo',
      message: 'Bar'
    })
    const Factory = () => 1
    ObjectStoreManager.add(mgr, errorInfoCb, Factory, 'id1')
    const error = _.attempt(
      ObjectStoreManager.add,
      mgr,
      errorInfoCb,
      Factory,
      'id1'
    )
    expect(error).toBeInstanceOf(Error)
    expect(error).toHaveProperty('name', 'Foo')
    expect(error).toMatchSnapshot()
  })

  test('It should throw error with information if item already exist | error containing spec will return and entity containing spec object', () => {
    const mgr = createManager()
    const errorInfoCb = () => ({
      name: 'Foo',
      message: 'Bar'
    })
    const Factory = () => {
      throw new Error('factory error')
    }

    const error = _.attempt(
      ObjectStoreManager.add,
      mgr,
      errorInfoCb,
      Factory,
      'id1',
      { spec: 'mySpec' }
    )

    expect(error).toBeInstanceOf(Error)
    expect(error).toHaveProperty('entityId', 'id1')
    expect(error).toMatchSnapshot()
  })

  test('It should return the object id and spec with no errors when ObjectStoreManager is called once', () => {
    const mgr = createManager()
    const errorInfoCb = () => ({
      name: 'Foo',
      message: 'Bar'
    })
    const Factory = (spec, id) => `${id}:${spec.type}`

    ObjectStoreManager.add(mgr, errorInfoCb, Factory, 'id1', { type: 'myType' })

    expect(mgr.store.get('id1')).toEqual('id1:myType')
  })
})
