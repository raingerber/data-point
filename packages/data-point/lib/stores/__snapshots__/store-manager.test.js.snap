// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add It should throw error with information if item already exist | error containing spec will return and entity containing spec object 1`] = `
[Error: factory error
Entity "id1": {
  "spec": "mySpec"
}]
`;

exports[`add It should throw error with information if item already exist 1`] = `[Foo: Bar]`;
